src/main.rs

```diff
@@ -1,3 +1,95 @@
+use rand::{rng, Rng};
+use std::cmp::Ordering;
+use std::io;
+
+const GENERATED_NUMBER_UPPER_LIMIT: i32 = 100;
+const GENERATED_NUMBER_LOWER_LIMIT: i32 = 1;
+
 fn main() {
-    println!("Hello, world!");
+    let generated_number: i32 = initialize_guessing_game();
+
+    run_guessing_game_loop(generated_number);
+}
+
+/// This function prints the game instructions to the user
+/// And randomly generates a number between 1-100 for the user to guess
+///
+/// # Returns
+///
+/// The generated number for the user to guess
+fn initialize_guessing_game() -> i32 {
```

Comment:
due to the scale of this program, this doesn't really matter. but I'm going to be really pedantic and say that this function does too many things.
The minute my function has the word `and` in it's description, I know it should be split into 2 functions.

I would do 

```rust
fn print_welcome_message() {
...
}
```

and also

```rust
fn generate_random_number() -> i32 {
...
}
```